scripts/components/CalendarAddEventForm.jsx

```diff
@@ -0,0 +1,167 @@
+      return;
+    }
+
+    Api.addPropertyCalendarEvents({
+      data: {
+        eventDesc: this.state.eventDesc,
+        date: this.state.date,
+        time: this.state.time,
+        notes: this.state.notes,
+        propertyId: propertyId,
+      },
+      callback: (err, response) => {
+        if (err) {
+          return;
+        }
+
+        // Clear the form
+        this.setState({
+          eventDesc: '',
+          date: '',
+          time: '',
+          notes: '',
+        });
+
+        this.props.eventAdded();
+        this.refs.dialog.dismiss();
+      }
+    });
+  },
+
+  // Validate the form, returns the Joi result of the validation.
+  validate() {
+    return Joi.validate({
+      'Event Description': this.state.eventDesc,
+      Date: this.state.date,
+      Time: this.state.time,
+      Notes: this.state.notes,
+    }, schema);
+   },
+
+  render() {
+    var { eventDesc, date, time, notes, validationError } = this.state;
+    var standardActions = [
+      { text: 'Cancel' },
+      { text: 'Add Event', onTouchTap: this.onSubmit, ref: 'submit' }
+    ];
+    var validationErrorMsg = validationError ? (
+      <JoiError error={validationError} fillBackground={true} />
+    ) : null;
+
+    return (
+      <div style={style.formContainer}>
+        <RaisedButton label="Add Event"
+                      primary={true}
+                      onClick={this.onButtonClick} />
+        <Dialog
+          title="Add an Event"
+          actions={standardActions}
+          actionFocus="submit"
+          modal="true"
+          ref="dialog">
+
+          {validationErrorMsg}
+
+          <TextField
+            value={eventDesc}
+            name="Event Description"
+            onChange={this.onChange.bind(this, 'eventDesc')}
+            floatingLabelText="Event Description"
+            hintText="Describe what will be happening" />
+          <br />
+          <TextField
+            value={date}
+            name="Date"
+            onChange={this.onChange.bind(this, 'date')}
+            floatingLabelText="Date"
+            hintText="Enter the date. (dd/mm/yyyy)" />
+          <TextField
+            value={time}
+            name="Time"
+            onChange={this.onChange.bind(this, 'time')}
+            floatingLabelText="Time"
+            hintText="Enter the time. (h:mm ampm)" />
+          <TextField
+            value={notes}
+            multiLine={true}
+            name="Notes"
+            onChange={this.onChange.bind(this, 'notes')}
+            floatingLabelText="Notes"
+            hintText="Add more detailed notes here"
+            fullWidth />
+        </Dialog>
+      </div>
+    );
+  }
+});
+
+var style = {
+
+};
+
+// Validation schema for user profile form data.
+var schema = Joi.object().keys({
+  'Event Description': Joi.string().required(),
+  Date: Joi.date().format('DD/MM/YYYY'),
+  Time: Joi.date().format('h:mm a'),
+  Notes: Joi.optional(),
```

Comment:
(Same comment on the server)
You have imposed a constraint in the DB schema on the length of both notes and eventDesc. You should reflect those constraints here. Simple case, if I write a big note with >1024 characters, everything greater than 1024 characters will simply be deleted. This is a general habit to be in when programming a system. If you introduce an artificial limit (eg 1024 characters) every part of the system should reflect this limit. Otherwise you just litter the system with silent failures and bugs, which over time gets really annoying. And these things are typically easy to prevent when you introduce these constraints. "Oh I just made a hard limit on how many characters you can add, better make sure the input validation lets someone know this is the case so it doesn't silently fail on them".